src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -333,14 +334,25 @@ pub struct CloudwatchKey {
 
 fn partition(
     event: Event,
-    group: &Bytes,
+    group: &Template,
     stream: &Template,
 ) -> Option<PartitionInnerBuffer<Event, CloudwatchKey>> {
+    let group = match group.render(&event) {
+        Ok(b) => b,
+        Err(missing_keys) => {
+            warn!(
+                message = "group keys do not exist on the event; dropping event.",
+                keys = ?missing_keys
+            );
+            return None;
+        }
+    };
+
     let stream = match stream.render(&event) {
         Ok(b) => b,
         Err(missing_keys) => {
             warn!(
-                message = "Keys do not exist on the event. Dropping event.",
+                message = "stream keys do not exist on the event; dropping event.",
```

Comment:
I'd probably prefer something like `keys in stream template`/`keys in group template` to `stream keys`/`group keys` for these two, just for clarity.